bind.go

```diff
@@ -111,11 +111,12 @@ func (b *DefaultBinder) Bind(i interface{}, c Context) (err error) {
 	if err := b.BindPathParams(c, i); err != nil {
 		return err
 	}
-	// Issue #1670 - Query params are binded only for GET/DELETE and NOT for usual request with body (POST/PUT/PATCH)
+	// Issue #1670 - Query params are binded only for GET/DELETE/HEAD and NOT for usual request with body (POST/PUT/PATCH)
```

Comment:
Better to rewrite the whole comment. I would suggest:
```
// Only bind query parameters for GET/DELETE/HEAD to avoid unexpected behavior with destination struct binding from body. For example a request URL `&id=1&lang=en` with body `{"id":100,"lang":"de"}` would lead to precedence issues.
	// The HTTP method check restore pre v4.1.11 behavior to avoid these problems (see issue #1670)
```

Reply:
Comment adjusted